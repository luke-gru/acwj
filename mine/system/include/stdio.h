#ifndef _STDIO_H_
#define _STDIO_H_

#include <stddef.h>
#include <stdarg.h>

#define EOF (-1)

// This FILE definition will do for now
typedef char *FILE;
extern FILE *stderr;
extern FILE *stdout;

FILE *fopen(char *pathname, char *mode);
size_t fread(void *ptr, size_t size, size_t nmemb, FILE *stream);
size_t fwrite(void *ptr, size_t size, size_t nmemb, FILE *stream);
int fclose(FILE *stream);
int printf(char *format);
int fprintf(FILE *stream, char *format);
size_t getline(char **lineptr, size_t *n, FILE *stream);

FILE *popen(const char *command, const char *type);
int pclose(FILE *stream);

int vfprintf(FILE *stream, const char *format, va_list ap);

#endif	// _STDIO_H_
