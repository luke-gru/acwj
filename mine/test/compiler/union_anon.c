#include <stdio.h>
struct S {
  union {
    int a;
    int b;
  };
  int c;
};

struct S s;

int main() {
  s.a = 10;
  s.b = 20;
  s.c = 40;
  printf("s.a: %d\n", s.a); // 20
  printf("s.b: %d\n", s.b); // 20
  printf("s.c: %d\n", s.c); // 40

  return (0);
}
