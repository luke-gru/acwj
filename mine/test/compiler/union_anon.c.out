s.a: 20
s.b: 20
s.c: 40
